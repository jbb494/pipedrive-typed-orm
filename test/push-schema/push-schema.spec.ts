import { sleep } from "bun";
import {
  afterEach,
  beforeAll,
  beforeEach,
  describe,
  expect,
  it,
} from "bun:test";
import { pushToPipedrive } from "src/push-schema";
import { getAllFields } from "src/push-schema/utils";
import {
  removeAllCustomFieldsPipedrive,
  removeDynamicFields,
} from "./push-schema.helper";

import { oneFieldSchemas } from "./schemas";

describe("Push schema", () => {
  beforeAll(async () => {
    const results = await removeAllCustomFieldsPipedrive();

    expect(results.ok).toEqual(true);
  });
  beforeEach(() => {
    process.env.SCHEMA_PATH = undefined;
  });
  afterEach(async () => {
    const results = await removeAllCustomFieldsPipedrive();

    expect(results.ok).toEqual(true);
  });

  describe("Testing fields", () => {
    Object.entries(oneFieldSchemas).map(([key, schema]) => {
      it(`Should push the ${key} schema to pipedrive (add one field to person and to lead)`, async () => {
        const results = await pushToPipedrive(schema);

        const resultDealState = await getAllFields("dealFields");
        sleep(100);
        const resultPersonState = await getAllFields("personFields");

        expect(resultDealState.ok).toBeTrue();
        expect(resultPersonState.ok).toBeTrue();

        expect(results.ok).toEqual(true);

        expect(results.value.resultLead.added).toEqual(1);
        expect(results.value.resultLead.removed).toEqual(0);
        expect(results.value.resultPerson.added).toEqual(1);
        expect(results.value.resultPerson.removed).toEqual(0);

        expect(
          resultDealState.value.data
            .filter((x: any) => x.edit_flag)
            .map(removeDynamicFields)
        ).toMatchSnapshot();
        expect(
          resultPersonState.value.data
            .filter((x: any) => x.edit_flag)
            .map(removeDynamicFields)
        ).toMatchSnapshot();
      });
    });
  });
  describe("Testing actions", () => {
    it("After applying a schema, the second time shouldnt add nor remove anything", async () => {
      const results = await pushToPipedrive(oneFieldSchemas.textSchema);

      expect(results.ok).toEqual(true);

      expect(results.value.resultLead.added).toEqual(1);
      expect(results.value.resultLead.removed).toEqual(0);
      expect(results.value.resultPerson.added).toEqual(1);
      expect(results.value.resultPerson.removed).toEqual(0);

      const resultsSecondTime = await pushToPipedrive(
        oneFieldSchemas.textSchema
      );

      expect(resultsSecondTime.ok).toEqual(true);

      expect(resultsSecondTime.value.resultLead.added).toEqual(0);
      expect(resultsSecondTime.value.resultLead.removed).toEqual(0);
      expect(resultsSecondTime.value.resultPerson.added).toEqual(0);
      expect(resultsSecondTime.value.resultPerson.removed).toEqual(0);
    });
    it("After applying a schema, the second should remove one and add one", async () => {
      const results = await pushToPipedrive(oneFieldSchemas.textSchema);

      expect(results.ok).toEqual(true);

      expect(results.value.resultLead.added).toEqual(1);
      expect(results.value.resultLead.removed).toEqual(0);
      expect(results.value.resultPerson.added).toEqual(1);
      expect(results.value.resultPerson.removed).toEqual(0);

      const resultsSecondTime = await pushToPipedrive(
        oneFieldSchemas.doubleSchema
      );

      expect(resultsSecondTime.ok).toEqual(true);

      expect(resultsSecondTime.value.resultLead.added).toEqual(1);
      expect(resultsSecondTime.value.resultLead.removed).toEqual(1);
      expect(resultsSecondTime.value.resultPerson.added).toEqual(1);
      expect(resultsSecondTime.value.resultPerson.removed).toEqual(1);
    });
  });
});
