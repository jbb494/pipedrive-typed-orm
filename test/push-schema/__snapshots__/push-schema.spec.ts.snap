// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Push schema Testing fields Should push the date schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the date schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the double schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the double schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enum schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enum schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the monetary schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the monetary schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the phone schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the phone schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the set schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the set schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the text schema to pipedrive 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the text schema to pipedrive 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the date schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the date schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the double schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the double schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enum schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enum schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the monetary schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the monetary schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the phone schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the phone schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the set schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the set schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the text schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the text schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the dateSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the dateSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "date",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "date",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the doubleSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the doubleSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "double",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "double",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enumSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the enumSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "enum",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "enum",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the monetarySchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the monetarySchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "monetary",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "monetary",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
  {
    "active_flag": true,
    "edit_flag": true,
    "field_type": "varchar",
    "id_suffix": "currency",
    "is_subfield": true,
    "mandatory_flag": false,
    "name": "Currency of monetary",
  },
]
`;

exports[`Push schema Testing fields Should push the phoneSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the phoneSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "phone",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "phone",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the setSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the setSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "set",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "set",
    "options": [
      {
        "label": "A",
      },
      {
        "label": "B",
      },
    ],
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the textSchema schema to pipedrive (add one field to person and to lead) 1`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "projects_detail_visible_flag": false,
    "searchable_flag": true,
    "show_in_pipelines": {
      "pipeline_ids": [],
      "show_in_all": true,
    },
    "sortable_flag": true,
    "update_time": null,
  },
]
`;

exports[`Push schema Testing fields Should push the textSchema schema to pipedrive (add one field to person and to lead) 2`] = `
[
  {
    "active_flag": true,
    "add_visible_flag": false,
    "bulk_edit_allowed": true,
    "description": null,
    "details_visible_flag": true,
    "edit_flag": true,
    "field_type": "text",
    "filtering_allowed": true,
    "group_id": null,
    "important_flag": false,
    "json_column_flag": true,
    "last_updated_by_user_id": 19039577,
    "mandatory_flag": false,
    "name": "text",
    "searchable_flag": true,
    "sortable_flag": true,
    "update_time": null,
  },
]
`;
